sdk/dynamicconfig/sdk/metric/controller/push/push.go

```diff
@@ -172,6 +198,34 @@ func (c *Controller) OnUpdatedConfig(config *dynamicconfig.Config) error {
 	return nil
 }
 
+// Iterate through the schedules for two purposes:
+//    -Return the minimal non-zero period
+//    -Reset lastCollected, set each CollectionPeriod's last collection
+//    timestamp to now. 
+func (c *Controller) updateFromSchedules() time.Duration {
+	now := time.Now()
+
+	var period pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod = 0
+	c.lastCollected = make(map[pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod]time.Time)
+
+	for _, schedule := range *c.schedules {
+		// If CollectionPeriod is 0, we do not do anything with it
+		if schedule.Period == 0 {
+			continue
+		}
+
+		if _, ok := c.lastCollected[schedule.Period]; !ok {
+			c.lastCollected[schedule.Period] = now
+		}
+
+		if period == 0 || period > schedule.Period {
```

Comment:
period < schedule.Period

Reply:
I think the original period > schedule.Period is correct. At the end we want period to reflect the smallest period seen in any of the schedules, so if we see a schedule with a period smaller than period, we replace period with it.

That said, this might be clearer if we rename "period" to "smallestPeriod" or "minimumPeriod".